Marlin/src/module/stepper.cpp

```diff
@@ -2856,6 +2915,80 @@ hal_timer_t Stepper::block_phase_isr() {
   return interval;
 }
 
+#if ENABLED(LA_ZERO_SLOWDOWN)
+
+  void Stepper::set_la_interval(const int32_t rate) {
+    if (rate == 0) { la_interval = LA_ADV_NEVER; return; }
+
+    const bool forward_e = rate > 0;
+    la_interval = calc_timer_interval(uint32_t(ABS(rate)) >> current_block->la_scaling);
+    if (forward_e == motor_direction(E_AXIS)) return;
+
+    last_direction_bits.toggle(E_AXIS);
+    count_direction.e = -count_direction.e;
+    DIR_WAIT_BEFORE();
+    E_APPLY_DIR(forward_e, false);
+    TERN_(FTM_OPTIMIZE_DIR_STATES, last_set_direction = last_direction_bits);
+    DIR_WAIT_AFTER();
+  }
+
+  hal_timer_t Stepper::zero_slowdown_isr() {
+    /**
+     * 1. [x] move to separate isr
+     * 2. [x] keep computations in movement step scale
+     * 3. [ ] precompute a_max in movement step scale and inter block correction factors in the planner
+     * 4. [ ] either use precomputed dt values or make update interval depend on la acceleration rate
+     * 5. [ ] optimise intermediate computations (e.g avoid dividing by dt to then multiply the variable by dt)
+     * 6. [ ] use int arithmetic (e.g bitshift instead of division/mult, fixed point, etc)
+     */
+    #define UPDATE_FREQ 4096 // hz
+    constexpr uint32_t interval = STEPPER_TIMER_RATE / UPDATE_FREQ;
+    constexpr float dt = float(interval) / float(STEPPER_TIMER_RATE);
+    constexpr float dt_inv = UPDATE_FREQ;
+    if (current_block) {
+      float target_la_step_count;
+      if (la_active) {
+        const float k = Planner::extruder_advance_K[E_INDEX_N(current_block->extruder)];
+        target_la_step_count = curr_step_rate * k;
+      }
+      else
+        target_la_step_count = 0; // (de)retration may start with non-zero current_la_step_rate and/or count. This needs to be gradually compensated for
+
+      const float distance_to_target = target_la_step_count - current_la_step_count,
+                  one_shot_v = distance_to_target * dt_inv;
+      float a = ABS(one_shot_v - current_la_step_rate) * dt_inv;
+      NOMORE(a, a_max);
+
+      float stopping_distance = current_la_step_rate * current_la_step_rate / (2 * a_max);
```

Comment:
Could this be written using `POW(current_la_step_rate, 2)` instead of `current_la_step_rate * current_la_step_rate`?

Reply:
Absolutely, is it more readable or also more efficient?

Reply:
I did not test it. I assume that the compiler will spit out the same assembler code in most cases. But `POW` explicitly relies on the HAL, so - depending on the arch - it _might_ be faster than doing `x * x` in some cases. But mostly, it just feels like the right command to use here 😉